docs/sources/configure-client/grafana-agent/go_pull.md

```diff
@@ -1,258 +1,224 @@
 ---
 title: "Go (pull mode)"
 menuTitle: "Go (pull mode)"
-description:  "Instrumenting Golang applications for continuous profiling"
+description: "Set up Go profiling in pull mode"
 weight: 10
 ---
 
 # Go (pull mode)
 
-The Grafana Agent is responsible for pulling profiles from applications and delivering them
-to the Pyroscope database.
+In pull mode, the Grafana Agent periodically retrieves profiles from Golang applications, specifically targeting the
+`/debug/pprof/*` endpoints.
 
-### Prerequisites
+## Set up Go profiling in pull mode
 
-To be able to pull profiles from applications, your applications needs to expose [pprof endpoints](https://pkg.go.dev/net/http/pprof).
+To set up Golang profiling in pull mode, you need to:
 
-Before proceeding with the configuration, ensure that you have the following:
+1. Expose pprof endpoints
+2. Install Grafana Agent
+3. Prepare Grafana Agent configuration file
+4. Start Grafana Agent
 
-1. Install Grafana Agent in [flow mode](/docs/agent/latest/flow/setup/install/)
-2. Configure Grafana Agent in flow mode
-3. Start the Grafana Agent
+### Expose pprof endpoints
 
-## Adding Profiling to the Grafana Agent
+Ensure your Golang application exposes pprof endpoints.
 
-This guide presents how to configure the Grafana Agent for scraping performance profiles from a pprof endpoint using `pyroscope.scrape` and then write to a reciever defined in `pyroscope.write`.
+1. Get `godeltaprof` package
 
-**While this documentation will contain a simple example for more configuration options visit the agent documentation for [`pyrosope.scrape`](/docs/agent/latest/flow/reference/components/pyroscope.scrape/) and [`pyroscope.write`](/docs/agent/latest/flow/reference/components/pyroscope.write/#pyroscopewrite).**
+    ```bash
+    go get github.com/grafana/pyroscope-go/godeltaprof@latest
+    ```
 
-## Agent Configuration
+2. Import `net/http/pprof` and `godeltaprof/http/pprof` packages at the start of your application.
 
-In [`/agent/config/config.river`](https://github.com/grafana/pyroscope/blob/main/examples/grafana-agent/agent/config/config.river) file you will need two blocks, `pyroscope.write` and `pyroscope.scrape`:
+    ```go
+    import _ "net/http/pprof"
+    import _ "github.com/grafana/pyroscope-go/godeltaprof/http/pprof"
+    ```
 
-![Grafana agent go pull diagram](/media/docs/pyroscope/grafana_agent_pull_mode_diagram.png)
+### Install Grafana Agent
 
-1. `pyroscope.write` to configure the receivers to which the profiles are forwarded.
-2. `pyroscope.scrape` to establish a pprof scraping job for specific targets. The performance profiles obtained from the scraping process are then passed to the receivers specified in `forward_to`. You can define multiple `pyroscope.scrape` components, each distinguished by unique labels.
+[//]: # (TODO&#40;korniltsev&#41; What should go here?)
```

Comment:
Are there specific things that someone has to do to Agent configuration to use Go in pull mode? If so, those things should be added here. 

At a minimum, we should link to the Agent installation doc. If we have more time, ideally we should include a config example for Agent for Pyroscope-specific config options. The latter would make the doc more usable. 

Here's another example. In the [Set up a test app doc in Tempo](https://grafana.com/docs/tempo/latest/setup/set-up-test-app/), we use an example Agent config as part of the procedure. (We have a PR that replaces that example with Flow instead of Static mode.) 

The minimum viable doc option here is a link agent installation procedures. We can always iterate and add more specifics. If this is the option you choose, you should make sure to create a doc issue so the update to add Agent config doesn't get lost.

Reply:
Ha! And underneath you have the configuration file! Never mind! :)